staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/types.go

```diff
@@ -67,5 +69,27 @@ type Key struct {
 	Secret string `json:"secret"`
 }
 
+// GKMSConfig contains API configuration for Google KMS transformer.
+type GKMSConfig struct {
+	// projectID is the GCP project which hosts the key to be used. It defaults to the GCP project
+	// in use, if you are running on Kubernetes on GKE/GCE. Setting this field will override
+	// the default. It is not optional if Kubernetes is not on GKE/GCE.
+	// +optional
+	ProjectID string `json:"projectID,omitempty"`
+	// location is the KMS location of the KeyRing to be used for encryption. The default value is "global".
+	// It can be found by checking the available KeyRings in the IAM UI.
+	// This is not the same as the GCP location of the project.
+	// +optional
+	Location string `json:"location,omitempty"`
+	// keyRing is the keyRing of the hosted key to be used. The default value is "google-kubernetes".
+	// +optional
+	KeyRing string `json:"keyRing,omitempty"`
+	// cryptoKey is the name of the key to be used for encryption of Data-Encryption-Keys.
+	CryptoKey string `json:"cryptoKey,omitempty"`
+	// cacheSize is the maximum number of secrets which are cached in memory. The default value is 1000.
+	// +optional
+	CacheSize int `json:"cacheSize,omitempty"`
+}
+
```

Comment:
Totally out of scope for this PR, but I would like moving this file to some other location to break a dependency cycle, and allow us to simply pass GKMSConfig objects to the constructors in `value/encrypt/kms`, instead of all the 5 fields separately.

Maybe `staging/src/k8s.io/api/storage/v1`, but am not sure.

I am hoping we can let all this (and the chunks of 5 arguments everywhere) be like this for the time being, and fix this in a new PR soonish.